src/app/catalog/recommendation/services/recommendation.service.spec.ts

```diff
@@ -17,8 +18,12 @@ describe('RecommendationService', () => {
 
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [RecommendationService],
-      imports: [HttpClientTestingModule],
+      imports: [],
+      providers: [
+        RecommendationService,
+        provideHttpClient(withInterceptorsFromDi()),
```

Comment:
I found this on the official Angular page for "withInterceptorsFromDi".
https://angular.dev/api/common/http/withInterceptorsFromDi?tab=description
Prefer [withInterceptors](https://angular.dev/api/common/http/withInterceptors) and functional interceptors instead, as support for DI-provided interceptors may be phased out in a later release.

Maybe you should check this.

Reply:
Good hint, thanks. Turns out that the withInterceptor...() can be removed completely in all places.